xlgui/widgets/playlist.py

```diff
@@ -858,10 +860,16 @@ def get_sort_by(self):
         sortcol = self.get_sort_column()
         if sortcol:
             reverse = sortcol.get_sort_order() == Gtk.SortType.DESCENDING
-            sort_by = [sortcol.name] + list(common.BASE_SORT_TAGS)
+            if not self._sort_columns:
+                sort_by = [sortcol.name] + list(common.BASE_SORT_TAGS)
+            else:
+                sort_by = self._sort_columns
```

Comment:
This seems wrong. What happens if the sort column didn't exist previously?

Reply:
If there was no sort column, would we not be in the other branch (as sortcol would be None)?

Reply:
What I meant was what if the column itself did not previously exist? Then it couldn't be in the _sort_columns list (which means it would get skipped here?)

Reply:
Is there a scenario where that happens? I originally tried to safe-guard against _sort_columns being empty in both branches because I was not sure if it could happen that get_sort_by() would be called and have a valid sortcol without on_column_clicked() happening first (which is where _sort_columns gets modified)...

I am asking because after revising the commit to initialize _sort_columns with ```list(common.BASE_SORT_TAGS)```, I also tried to simplify that part so that _sort_columns is always returned and sortcol is used only to get sorting direction. As far as I can tell, sortcol can be valid only if user clicked on a column and on_column_clicked() was executed, because that is where sort indicator gets set (and that is what get_sort_column() in get_sort_by() is checking for).